lib/posix/Kconfig

```diff
@@ -56,6 +56,13 @@ config MAX_PTHREAD_KEY_COUNT
 	help
 	  Maximum number of simultaneously active keys in a POSIX application.
 
+config MAX_PTHREAD_BARRIER_COUNT
+	int "Maximum simultaneously active barriers in a POSIX application"
+	default 5
```

Comment:
I would very much like to reduce this default to 1 for all of the pthread object pools at some point.

A later ticket.